llm/prompt.go

```diff
@@ -68,27 +68,58 @@ type Response struct {
 }
 
 func getProvider(ctx context.Context) LLMProvider {
-	if key := os.Getenv("GEMINI_API_KEY"); key != "" {
-		provider := NewGeminiProvider(key, ctx)
-		if provider != nil {
-			return provider
-		}
+	// Define provider configurations
+	providerConfigs := map[string]struct {
+		envKey string
+		initFn func(string, context.Context) LLMProvider
+	}{
+		"gemini": {
+			envKey: "GEMINI_API_KEY",
+			initFn: func(key string, ctx context.Context) LLMProvider {
+				return NewGeminiProvider(key, ctx)
+			},
+		},
+		"openai": {
+			envKey: "OPENAI_API_KEY",
+			initFn: func(key string, ctx context.Context) LLMProvider {
+				return NewOpenaiProvider(key)
+			},
+		},
+		"anthropic": {
+			envKey: "ANTHROPIC_API_KEY",
+			initFn: func(key string, ctx context.Context) LLMProvider {
+				return NewAnthropicProvider(key)
+			},
+		},
 	}
 
-	if key := os.Getenv("OPENAI_API_KEY"); key != "" {
-		provider := NewOpenaiProvider(key)
-		if provider != nil {
-			return provider
+	configuredProvider := viper.GetString("provider")
+	
+	providersToTry := []string{"gemini", "openai", "anthropic"}
+	if configuredProvider != "" {
+		if _, exists := providerConfigs[configuredProvider]; !exists {
+			fmt.Fprintf(os.Stderr, "Error: Unknown provider '%s' specified in config\n", configuredProvider)
+			return nil
 		}
+		providersToTry = []string{configuredProvider}
 	}
 
-	if key := os.Getenv("ANTHROPIC_API_KEY"); key != "" {
-		provider := NewAnthropicProvider(key)
-		if provider != nil {
-			return provider
+	for _, providerName := range providersToTry {
+		config := providerConfigs[providerName]
+		if key := os.Getenv(config.envKey); key != "" {
```

Comment:
What if we push getenv down to NewXXXProvider

At the end of the day they're supposed to know which keys they need, it's coupled in a way